src/extensions/wcf/Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater/WCFUpdateStep.cs

```diff
@@ -0,0 +1,226 @@
+            }
+
+            try
+            {
+                return Apply();
+            }
+            catch (Exception e)
+            {
+                Logger.LogError(e, "Unexpected error happened during CoreWCF porting.");
+                return Task.FromResult(new UpgradeStepApplyResult(UpgradeStepStatus.Failed, "Project failed to port to CoreWCF. Please review error message and log file for more information."));
+            }
+        }
+
+        public Task<UpgradeStepInitializeResult> Initialize(IProject project)
+        {
+            _path = new Dictionary<string, IEnumerable<string>>();
+            FindPath(project);
+            if (_path.Count != 3 && _path.Count != 4)
+            {
+                return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, "Cannot find required file(s) for WCF update. The project is not applicable for automated update and this step is complete.", BuildBreakRisk.Low));
+            }
+
+            // if the project is not applicable, skip the update step
+            if (!WCFUpdateChecker.IsWCFUpdateApplicable(_path, Logger))
+            {
+                var message = "This project is not applicable for updating to CoreWCF since references to WCF were not found. No more work needs to be done and this step is complete.";
+                Logger.LogInformation(message);
+                return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, message, BuildBreakRisk.Low));
+            }
+
+            // construct updaters
+            _configUpdater = UpdaterFactory.GetConfigUpdater(_path["config"].First(), Logger);
+            _packageUpdater = UpdaterFactory.GetPackageUpdater(_path["proj"].First(), Logger);
+            if (_path.ContainsKey("directives"))
+            {
+                _directiveUpdaters = UpdaterFactory.GetDirectiveUpdaters(_path["directives"], Logger);
+            }
+
+            if (_configUpdater is not null)
+            {
+                var configContext = UpdateRunner.GetContext(_configUpdater);
+                _sourceCodeUpdater = UpdaterFactory.GetSourceCodeUpdater(_path["main"].First(), configContext, Logger);
+            }
+
+            if (_configUpdater is null || _packageUpdater is null || _sourceCodeUpdater is null || (_path.ContainsKey("directives") && _directiveUpdaters is null))
+            {
+                Logger.LogWarning("Unexpected error happened when trying to construct the updaters. Please review error message and log.");
+                return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Failed, "Updaters cannot be constructed. Please review error message and log for more information.", BuildBreakRisk.Medium));
+            }
+
+            return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Incomplete, "WCF updaters are initialized. Ready to update.", BuildBreakRisk.None));
```

Comment:
The BuildBreakRisk is probably still medium here. That means that if the user proceeds with applying the change there's a "medium" chance that they'll need to go back and make manual updates afterwards to get the project building again.